File: backend/src/main/java/dev/equalcoding/flashcards/models/FlashCardTag.java
package dev.equalcoding.flashcards.models;


import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "flash-card-tags")
public class FlashCardTag {
    @Id
    String tag;
    int count;
    int correctCount;
    int incorrectCount;
    int cardsSeenCount;

    public FlashCardTag(String tag, int count, int correctCount, int incorrectCount, int cardsSeenCount) {
        this.tag = tag;
        this.count = count;
        this.correctCount = correctCount;
        this.incorrectCount = incorrectCount;
        this.cardsSeenCount = cardsSeenCount;
    }



    public int getCorrectCount() {
        return correctCount;
    }

    public void setCorrectCount(int correctCount) {
        this.correctCount = correctCount;
    }

    public int getIncorrectCount() {
        return incorrectCount;
    }

    public void setIncorrectCount(int incorrectCount) {
        this.incorrectCount = incorrectCount;
    }

    public int getCardsSeenCount() {
        return cardsSeenCount;
    }

    public void setCardsSeenCount(int cardsSeenCount) {
        this.cardsSeenCount = cardsSeenCount;
    }

    public int getCount() {
        return count;
    }

    public void setCount(int count) {
        this.count = count;
    }

    public FlashCardTag() {
    }

    public String getTag() {
        return tag;
    }

    public void setTag(String tag) {
        this.tag = tag;
    }
}

File: frontend/lib/models/flash_card_tag.dart
class FlashCardTag {
  String? tag;
  int? count;
  int? correctCount;
  int? incorrectCount;
  int? cardsSeenCount;
  int? uniqueCardsSeenCount;


  FlashCardTag({this.tag, this.count, this.correctCount, this.incorrectCount,
      required this.cardsSeenCount, this.uniqueCardsSeenCount});

  FlashCardTag.fromJson(Map<String, dynamic> json) {
    tag = json['tag'];
    count = json['count'];
    correctCount = json['correctCount'];
    incorrectCount = json['incorrectCount'];
    cardsSeenCount = json['cardsSeenCount'];
    uniqueCardsSeenCount = json['uniqueCardsSeenCount'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['tag'] = tag;
    data['count'] = count;
    data['correctCount'] = correctCount;
    data['incorrectCount'] = incorrectCount;
    data['cardsSeenCount'] = cardsSeenCount;
    data['uniqueCardsSeenCount'] = uniqueCardsSeenCount;
    return data;
  }
}